libdnf-cli/output/changelogs.cpp

```diff
@@ -71,9 +65,12 @@ void print_changelogs(
             });
 
         // filter changelog
-        if (upgrades) {
+        if (filter.first == ChangelogFilterType::UPGRADES) {
             // Find the newest changelog on the installed version of the
             // package, and filter out any changelogs newer than that
+            auto & installed = std::get<libdnf::rpm::PackageQuery>(filter.second);
+            installed.filter_installed();
```

Comment:
The query is now created for each srpm, although it's the same for all of them. Can it be moved out of the loop?

Reply:
Oh, I will move the filtering out of the loop. Thanks!